opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -1429,6 +1429,19 @@ void CpGrid::addLgrsUpdateLeafView(const std::vector<std::array<int,3>>& cells_p
                                    const std::vector<std::array<int,3>>& endIJK_vec,
                                    const std::vector<std::string>& lgr_name_vec)
 {
+    // Check grid is Cartesian
+    const std::array<int,3>& coarseGrid_dim =  (*data_[0]).logical_cartesian_size_;
+    long unsigned int coarseGridXYZ = coarseGrid_dim[0]*coarseGrid_dim[1]*coarseGrid_dim[2];
+    if ((*data_[0]).global_cell_.size() != coarseGridXYZ){
+        if (comm().rank()==0){
+            OPM_THROW(std::logic_error, "Grid is not Cartesian. This type of refinement is not supported yet.");
+        }
+        else {
+            OPM_THROW_NOLOG(std::logic_error, "Grid is not Cartesian. This type of refinement is not supported yet.");
+        }
+    }
```

Comment:
I should have thought about this and mentioned it before:
We should move this check below the next one (lines 1445-1452). Then we are sure loadBalance was not called in a parallel run. 
The check should only be done on rank 0:
```
 if (comm().rank()==0){
    if ((*data_[0]).global_cell_.size() != coarseGridXYZ){
            OPM_THROW(std::logic_error, "Grid is not Cartesian. This type of refinement is not supported yet.");
        }
    }
```
As the other processes will have an empty grid. (`(*data_[0]).global_cell_.size()==0`).